src/http/server/handlers/static_file_handler.cr

```diff
@@ -99,15 +101,81 @@ class HTTP::StaticFileHandler
         end
       end
 
-      context.response.content_length = file_info.size
       File.open(file_path) do |file|
-        IO.copy(file, context.response)
+        if range_header = context.request.headers["Range"]?
+          range_header = range_header.lchop?("bytes=") || return context.response.respond_with_status :range_not_satisfiable
+          ranges = parse_ranges(range_header, file_info.size) || return context.response.respond_with_status :bad_request
+
+          # If any of the ranges start beyond the end of the file, we return an
+          # HTTP 416 Range Not Satisfiable.
+          # See https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests#partial_request_responses
+          if ranges.any? { |range| range.begin > file_info.size }
```

Comment:
[The success condition is "less than"](https://www.rfc-editor.org/rfc/rfc9110.html#section-14.1.2-11.1):

```suggestion
          # See https://www.rfc-editor.org/rfc/rfc9110.html#section-14.1.2-11.1
          if ranges.any? { |range| range.begin >= file_info.size }
```